File: src/main/java/com/example/kyudo_app/domain/model/GameMatch.java
package com.example.kyudo_app.domain.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Setter
@Getter
@Entity
@Table(name="game_match")
public class GameMatch {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name ="match_id")
    private Long matchId;
    @Column(name ="match_date")
    private Date matchDate;

    @ManyToMany
    @JoinTable(
        name = "GameMatch_User",  // 中間テーブルの名前
        joinColumns = @JoinColumn(name = "match_id"),  // GameMatchの外部キー
        inverseJoinColumns = @JoinColumn(name = "user_id")  // Userの外部キー
    )
    private List<User> users;  // GameMatchに関連する複数のユーザー
    @ManyToOne
    @JoinColumn(name = "Place_id", referencedColumnName = "Place_id")
    private Place place;
}

File: src/main/kotlin/com/example/kyudo_app/infrastructure/persistence/GameMatch.kt
package com.example.kyudo_app.infrastructure.persistence

import jakarta.persistence.*
import lombok.Getter
import lombok.Setter
import java.util.*

@Setter
@Getter
@Entity
@Table(name = "game_match")
class GameMatch {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "match_id")
    private var matchId: Long? = null

    @Column(name = "match_date")
    private var matchDate: Date? = null

    @ManyToMany
    @JoinTable(
        name = "GameMatch_User",
        joinColumns = [JoinColumn(name = "match_id")],
        inverseJoinColumns = [JoinColumn(name = "user_id")]
    )
    private val users: MutableList<User?>? = null // GameMatchに関連する複数のユーザー

    @ManyToOne
    @JoinColumn(name = "Place_id", referencedColumnName = "Place_id")
    private val place: Place? = null
}
